packages/flutter_tools/test/integration.shard/test_utils.dart

```diff
@@ -47,10 +47,12 @@ Future<void> getPackages(String folder) async {
     'get',
   ];
   final Process process = await processManager.start(command, workingDirectory: folder);
+  final StringBuffer output = StringBuffer();
   final StringBuffer errorOutput = StringBuffer();
+  process.stdout.transform(utf8.decoder).listen(output.write);
   process.stderr.transform(utf8.decoder).listen(errorOutput.write);
   final int exitCode = await process.exitCode;
   if (exitCode != 0) {
-    throw Exception('flutter pub get failed: $errorOutput');
+    throw Exception('flutter pub get failed: $errorOutput\n$output');
```

Comment:
The stdout and stderr might not have been completely drained at this point. But this change is fine if it is okay for the output here to be partial.

Reply:
I guess it'd be better for them to be complete! I presume a fix would be to `await Future.when([process.stdout, process.stderr]);` before throwing?

Reply:
You can capture the stream subscriptions from the `listen()` calls and use `asFuture()`
```dart
final StreamSubscription<String> stdoutSubscription = process.stdout.transform(utf8.decoder).listen(output.write);
final StreamSubscription<String> stderrSubscription = process.stderr.transform(utf8.decoder).listen(errorOutput.write);
...
await Future.wait([stdoutSubscription.asFuture<void>(), stderrSubscription.asFuture<void>()]);
```

https://api.dartlang.org/stable/2.5.1/dart-async/StreamSubscription/asFuture.html

Reply:
Thanks - I knew there was an `asFuture()` somewhere but I was looking in the wrong place!

Anyway, when fixing this up I realised this wasn't doing anything more than `processManager.run()` but was much more code, so I've changed it to use that.

I can't actually repro the original issue (which was causing timeouts - undoing this change did not bring back the timeouts on my Windows PC), however since the code here seems flawed anyway (it wasn't draining stderr either) it seems worthwhile fixing up (and simplifying).